File: tests/06_device/test06_device.c
#include <brisbane/brisbane.h>
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char** argv) {
  brisbane_init(&argc, &argv, true);

  char vendor[64];
  char name[64];
  int type;
  int nplatforms = 0;
  int ndevs = 0;
  brisbane_platform_count(&nplatforms);
  for (int i = 0; i < nplatforms; i++) {
    size_t size;
    brisbane_platform_info(i, brisbane_name, name, &size);
    printf("platform[%d] name[%s]\n", i, name);
  }

  brisbane_device_count(&ndevs);
  for (int i = 0; i < ndevs; i++) {
    size_t size;
    brisbane_device_info(i, brisbane_vendor, vendor, &size);
    brisbane_device_info(i, brisbane_name, name, &size);
    brisbane_device_info(i, brisbane_type, &type, &size);
    printf("dev[%d] vendor[%s] name[%s] type[0x%x]\n", i, vendor, name, type);
  }
  brisbane_finalize();

  return 0;
}

File: tests/06_device/test06_device.py
#!/usr/bin/env python

import brisbane

brisbane.init(True)

nplatforms = brisbane.platform_count()
for i in range(nplatforms):
  name = brisbane.platform_info(i, brisbane.brisbane_name)
  print "platform[", i, "] name[", name, "]"

ndevs = brisbane.device_count()
for i in range(ndevs):
  vendor = brisbane.device_info(i, brisbane.brisbane_vendor)
  name = brisbane.device_info(i, brisbane.brisbane_name)
  print "device[", i, "] vendor[", vendor, "] name[", name, "]"

brisbane.finalize()

